### DISCUSSION TITLE
How to alter the axis width in plotIPDF?

### DISCUSSION THREAD

**Discussion by Matuanz** [mtex-toolbox/mtex#2031] [Ask Anything] [answered]
Thanks for reading this question. 
I want to plot the inverse pole figure for some grains using plotIPDF function, but the width of axis is somehow too narrow, how to change the default value of the linewidth? 
I tried to use the commands:
<
plofIPDF(ebsd, ebsd.orientations, 'linewidth', 3)
>
But it seems only changed the width of the speckles, not the axis. 
And one more question, the top right corner is not closed, is this a bug? I don't know how to fix this either.
The first figure shows the figure I can get via the commands, and the second figure is what I really want to plot, can some one tell me how to do this? 
![Grain No 150  170-IPF](https://github.com/mtex-toolbox/mtex/assets/105851046/2e904fc4-43bc-4281-bc51-5a9580aaba34)
![2271704781949_ pic](https://github.com/mtex-toolbox/mtex/assets/105851046/769befe7-3f48-40ce-9245-8ef602380d9f)

Thank you.

**Comment by kilir** [comment, marked as answer]
Hi
a) plotIPDF takes orientations as the first argument, see help plotIPDF
b) to manipulate the graphics objects of the plot, those are for example in the children of the hgt transform of the sphericalPlot:
```matlab
plotIPDF(ebsd('q').orientations,xvector,'contourf')

f = gcm; %get mtexfigure handle
sp = getappdata(f.gca,'sphericalPlot') %get handle of sphericalPlot
for i=1:length(sp.hgt.Children)
if isgraphics(sp.hgt.Children(i),'Line')
    sp.hgt.Children(i).LineWidth=3
    sp.hgt.Children(i).Color=rand(3,1)
end
end
```
c) I don't get the gap, but I guess it is a rounding/cutoff problem in sphericalRegion/plot. You can try to decrease the step width of omega [l.27](https://github.com/mtex-toolbox/mtex/blob/72a4796aa5f5dc92dc6b6dcd56faf3130ded94ad/geometry/%40sphericalRegion/plot.m#L27) and see if this changes

Hope that helps.

Cheers,
Rüdiger

**Reply by Matuanz** [reply]
Thank you kilir, 
I can get access to the f.children.linewidth attribute following your code. But I don't know how to change the value for Axis width. For my case, I assume that I need to get handle of ipdfPlot by command 
'sp = getappdata(f.gca, 'ipdfPlot')',
but it doesn't work for me. 
Could you give me more detailed instructions?
Thanks alot !

**Reply by kilir** [reply]
Just use the code provided in b).

**Reply by Matuanz** [reply]
Hi kilir,
Thank you very much for your code. It totally works. 
Could you tell me how did you find out the way to do this? Or where to find the reference for those children/parent attributes?
In case I may have further needs of customising the plots, where should I find the information?
For example, if I type:
plotIPDF(ebsd.orientations, zvector);
The Title of the figure is 'Z', how can I change it to a custom name?

**Reply by kilir** [reply]
The title is set via 'mtexTitle()'.
with respect to the properties of the figure, the individual axes etc. you can simply browse those e.g. have a look what fields you'lll see in for example 'sp = getappdata(f.gca,'sphericalPlot')' from the above.
